assume/world.py

```diff
@@ -328,7 +312,12 @@ def add_unit_operator(self, id: str) -> None:
         if self.unit_operators.get(id):
             raise ValueError(f"Unit operator {id} already exists")
 
-        units_operator = UnitsOperator(available_markets=list(self.markets.values()))
+        units_operator = UnitsOperator(
+            simulation_start=self.start,
+            simulation_end=self.end,
+            available_markets=list(self.markets.values()),
+            train_freq=self.learning_config.get("train_freq", 24),
```

Comment:
I do not like the way the tran_freq is now handeled by the unit operator as it would be technically the task of the learning role. Just definition wise. Is there a way to keep it in the learning role somehow? Maybe we can discuss that?

Reply:
@kim-mskw I have now moved the definition of the task into the world setup itself. Please take a look at the new version

Reply:
yeah it makes more sense there somehow than in the unit operator. The tidiest way would still be in the learning role, but I see why that is not possible